  <main role="main">    <h1>Class ServerCloneException</h1>   <ul class="inheritance"> <li><a href="../../../../java.base/java/lang/object">java.lang.Object</a></li> <li> <ul class="inheritance"> <li><a href="../../../../java.base/java/lang/throwable">java.lang.Throwable</a></li> <li> <ul class="inheritance"> <li><a href="../../../../java.base/java/lang/exception">java.lang.Exception</a></li> <li> <ul class="inheritance"> <li><a href="../../../../java.base/java/lang/clonenotsupportedexception">java.lang.CloneNotSupportedException</a></li> <li> <ul class="inheritance"> <li>java.rmi.server.ServerCloneException</li> </ul> </li> </ul> </li> </ul> </li> </ul> </li> </ul>   <dl> <dt>All Implemented Interfaces:</dt> <dd><code><a href="../../../../java.base/java/io/serializable">Serializable</a></code></dd> </dl>  <pre data-language="java">public class ServerCloneException
extends CloneNotSupportedException</pre> <div>
<p>A <code>ServerCloneException</code> is thrown if a remote exception occurs during the cloning of a <code>UnicastRemoteObject</code>. </p>
<p>As of release 1.4, this exception has been retrofitted to conform to the general purpose exception-chaining mechanism. The "nested exception" that may be provided at construction time and accessed via the public <a href="#detail"><code>detail</code></a> field is now known as the <i>cause</i>, and may be accessed via the <a href="../../../../java.base/java/lang/throwable#getCause()"><code>Throwable.getCause()</code></a> method, as well as the aforementioned "legacy field." </p>
<p>Invoking the method <a href="../../../../java.base/java/lang/throwable#initCause(java.lang.Throwable)"><code>Throwable.initCause(Throwable)</code></a> on an instance of <code>ServerCloneException</code> always throws <a href="../../../../java.base/java/lang/illegalstateexception"><code>IllegalStateException</code></a>.</p>
</div> <dl> <dt>Since:</dt> <dd>1.1</dd> <dt>See Also:</dt> <dd>
<a href="unicastremoteobject#clone()"><code>UnicastRemoteObject.clone()</code></a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/serialized-form.html#java.rmi.server.ServerCloneException">Serialized Form</a>
</dd> </dl>      <section role="region">  <h2 id="field.summary">Fields</h2> <table>  <tr> <th scope="col">Modifier and Type</th> <th scope="col">Field</th> <th scope="col">Description</th> </tr> <tr> <td><code><a href="../../../../java.base/java/lang/exception">Exception</a></code></td> <th scope="row"><code><a href="#detail">detail</a></code></th> <td> <p>The cause of the exception.</p> </td> </tr> </table>  </section>  <section role="region">  <h2 id="constructor.summary">Constructors</h2> <table>  <tr> <th scope="col">Constructor</th> <th scope="col">Description</th> </tr> <tr> <th scope="row"><code><a href="#%3Cinit%3E(java.lang.String)">ServerCloneException</a>​(<a href="../../../../java.base/java/lang/string">String</a> s)</code></th> <td> <p>Constructs a <code>ServerCloneException</code> with the specified detail message.</p> </td> </tr> <tr> <th scope="row"><code><a href="#%3Cinit%3E(java.lang.String,java.lang.Exception)">ServerCloneException</a>​(<a href="../../../../java.base/java/lang/string">String</a> s,
                    <a href="../../../../java.base/java/lang/exception">Exception</a> cause)</code></th> <td> <p>Constructs a <code>ServerCloneException</code> with the specified detail message and cause.</p> </td> </tr> </table>  </section>  <section role="region">  <h2 id="method.summary">Methods</h2> <table>  <tr> <th scope="col">Modifier and Type</th> <th scope="col">Method</th> <th scope="col">Description</th> </tr> <tr id="i0"> <td><code><a href="../../../../java.base/java/lang/throwable">Throwable</a></code></td> <th scope="row"><code><a href="#getCause()">getCause</a>()</code></th> <td> <p>Returns the cause of this exception.</p> </td> </tr> <tr id="i1"> <td><code><a href="../../../../java.base/java/lang/string">String</a></code></td> <th scope="row"><code><a href="#getMessage()">getMessage</a>()</code></th> <td> <p>Returns the detail message, including the message from the cause, if any, of this exception.</p> </td> </tr> </table>  <h2 id="methods.inherited.from.class.java.lang.Throwable">Methods declared in class java.lang.<a href="../../../../java.base/java/lang/throwable">Throwable</a>
</h2> <code><a href="../../../../java.base/java/lang/throwable#addSuppressed(java.lang.Throwable)">addSuppressed</a>, <a href="../../../../java.base/java/lang/throwable#fillInStackTrace()">fillInStackTrace</a>, <a href="../../../../java.base/java/lang/throwable#getLocalizedMessage()">getLocalizedMessage</a>, <a href="../../../../java.base/java/lang/throwable#getStackTrace()">getStackTrace</a>, <a href="../../../../java.base/java/lang/throwable#getSuppressed()">getSuppressed</a>, <a href="../../../../java.base/java/lang/throwable#initCause(java.lang.Throwable)">initCause</a>, <a href="../../../../java.base/java/lang/throwable#printStackTrace()">printStackTrace</a>, <a href="../../../../java.base/java/lang/throwable#printStackTrace(java.io.PrintStream)">printStackTrace</a>, <a href="../../../../java.base/java/lang/throwable#printStackTrace(java.io.PrintWriter)">printStackTrace</a>, <a href="../../../../java.base/java/lang/throwable#setStackTrace(java.lang.StackTraceElement%5B%5D)">setStackTrace</a>, <a href="../../../../java.base/java/lang/throwable#toString()">toString</a></code>  <h2 id="methods.inherited.from.class.java.lang.Object">Methods declared in class java.lang.<a href="../../../../java.base/java/lang/object">Object</a>
</h2> <code><a href="../../../../java.base/java/lang/object#clone()">clone</a>, <a href="../../../../java.base/java/lang/object#equals(java.lang.Object)">equals</a>, <a href="../../../../java.base/java/lang/object#finalize()">finalize</a>, <a href="../../../../java.base/java/lang/object#getClass()">getClass</a>, <a href="../../../../java.base/java/lang/object#hashCode()">hashCode</a>, <a href="../../../../java.base/java/lang/object#notify()">notify</a>, <a href="../../../../java.base/java/lang/object#notifyAll()">notifyAll</a>, <a href="../../../../java.base/java/lang/object#wait()">wait</a>, <a href="../../../../java.base/java/lang/object#wait(long)">wait</a>, <a href="../../../../java.base/java/lang/object#wait(long,int)">wait</a></code>  </section>   <div>   <section role="region">  <h2 id="field.detail">Fields</h2>   <h3 id="detail">detail</h3> <pre data-language="java">public Exception detail</pre> <div>
<p>The cause of the exception. </p>
<p>This field predates the general-purpose exception chaining facility. The <a href="../../../../java.base/java/lang/throwable#getCause()"><code>Throwable.getCause()</code></a> method is now the preferred means of obtaining this information.</p>
</div>   </section>  <section role="region">  <h2 id="constructor.detail">Constructors</h2>   <h3 id="&lt;init&gt;(java.lang.String)">ServerCloneException</h3> <pre data-language="java">public ServerCloneException(String s)</pre> <p>Constructs a <code>ServerCloneException</code> with the specified detail message.</p> <dl> <dt>Parameters:</dt> <dd>
<code>s</code> - the detail message.</dd> </dl>    <h3 id="&lt;init&gt;(java.lang.String,java.lang.Exception)">ServerCloneException</h3> <pre data-language="java">public ServerCloneException(String s,
                            Exception cause)</pre> <p>Constructs a <code>ServerCloneException</code> with the specified detail message and cause.</p> <dl> <dt>Parameters:</dt> <dd>
<code>s</code> - the detail message.</dd> <dd>
<code>cause</code> - the cause</dd> </dl>   </section>  <section role="region">  <h2 id="method.detail">Methods</h2>   <h3 id="getMessage()">getMessage</h3> <pre data-language="java">public String getMessage()</pre> <p>Returns the detail message, including the message from the cause, if any, of this exception.</p> <dl> <dt>Overrides:</dt> <dd>
<code><a href="../../../../java.base/java/lang/throwable#getMessage()">getMessage</a></code> in class <code><a href="../../../../java.base/java/lang/throwable">Throwable</a></code>
</dd> <dt>Returns:</dt> <dd>the detail message</dd> </dl>    <h3 id="getCause()">getCause</h3> <pre data-language="java">public Throwable getCause()</pre> <p>Returns the cause of this exception. This method returns the value of the <a href="#detail"><code>detail</code></a> field.</p> <dl> <dt>Overrides:</dt> <dd>
<code><a href="../../../../java.base/java/lang/throwable#getCause()">getCause</a></code> in class <code><a href="../../../../java.base/java/lang/throwable">Throwable</a></code>
</dd> <dt>Returns:</dt> <dd>the cause, which may be <code>null</code>.</dd> <dt>Since:</dt> <dd>1.4</dd> </dl>   </section>  </div>  </main>  <div class="_attribution">
  <p class="_attribution-p">
    &copy; 1993, 2020, Oracle and/or its affiliates. All rights reserved.<br>Documentation extracted from Debian's OpenJDK Development Kit package.<br>Licensed under the GNU General Public License, version 2, with the Classpath Exception.<br>Various third party code in OpenJDK is licensed under different licenses (see Debian package).<br>Java and OpenJDK are trademarks or registered trademarks of Oracle and/or its affiliates.<br>
    <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.rmi/java/rmi/server/ServerCloneException.html" class="_attribution-link">https://docs.oracle.com/en/java/javase/11/docs/api/java.rmi/java/rmi/server/ServerCloneException.html</a>
  </p>
</div>
